addons/minio/template/base/install.sh

```diff
@@ -247,7 +271,344 @@ function allow_pvc_resize() {
+            continue
+        fi
+
+        if [ "$space_needed" -gt "$free_space" ]; then
+            printf "Not enough space to migrate %s bytes from %s to %s\n" "$space_needed" "$MINIO_HOSTPATH" "$migration_path"
+            continue
+        fi
+
+        break
+    done
+
+    local suffix
+    suffix=$(echo $RANDOM | md5sum | head -c 8)
+    local migration_path
+    migration_path=$(printf "%s/minio-migration-%s" "$PROMPT_RESULT" "$suffix")
+    mkdir "$migration_path"
+    MINIO_MIGRATION_HOSTPATH=$migration_path
+}
+
+# minio_create_fs_migration_deployment creates a minio deployment and make it available through a service called
+# 'minio-migrate-fs-backend'. this new deployment uses the same credentials used by the original minio
+# deployment. if the installation uses host path and not a pvc, a temporary path is requested from the user.
+function minio_create_fs_migration_deployment() {
+    local src="$DIR/addons/minio/__MINIO_DIR_NAME__/migrate-fs"
+    local dst="$DIR/kustomize/minio/migrate-fs"
+
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        mkdir -p "$dst/hostpath"
+        render_yaml_file_2 "$src/hostpath/deployment.yaml" > "$dst/hostpath/deployment.yaml"
+        render_yaml_file_2 "$src/hostpath/kustomization.yaml" > "$dst/hostpath/kustomization.yaml"
+        render_yaml_file_2 "$src/hostpath/service.yaml" > "$dst/hostpath/service.yaml"
+        kubectl apply -k "$dst/hostpath/"
+    else
+        mkdir -p "$dst/pvc"
+        render_yaml_file_2 "$src/pvc/deployment.yaml" > "$dst/pvc/deployment.yaml"
+        render_yaml_file_2 "$src/pvc/kustomization.yaml" > "$dst/pvc/kustomization.yaml"
+        render_yaml_file_2 "$src/pvc/service.yaml" > "$dst/pvc/service.yaml"
+        render_yaml_file_2 "$src/pvc/pvc.yaml" > "$dst/pvc/pvc.yaml"
+        kubectl apply -k "$dst/pvc/"
+    fi
+
+    local endpoint
+    endpoint=$(kubectl -n ${MINIO_NAMESPACE} get service minio-migrate-fs-backend | tail -n1 | awk '{ print $3}')
+
+    printf "Awaiting minio fs migration readiness\n"
+    if ! spinner_until 120 minio_ready "$endpoint"; then
+        bail "Minio FS Migration API failed to report healthy"
+    fi
+}
+
+# minio_destroy_fs_migration_deployment deletes the temporary minio deployment used to migrate the object
+# storage.
+function minio_destroy_fs_migration_deployment() {
+    local dst="$DIR/kustomize/minio/migrate-fs"
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        kubectl delete -k "$dst/hostpath"
+        return
+    fi
+    kubectl delete -k "$dst/pvc"
+}
+
+# minio_swap_fs_migration_pvs swaps the pv backing the minio deployment with the pv used during the migration.
+function minio_swap_fs_migration_pvs() {
+    local minio_pv=$1
+    local migration_pv=$2
+    local src="$DIR/addons/minio/__MINIO_DIR_NAME__"
+    local dst="$DIR/kustomize/minio"
+
+    MINIO_ORIGINAL_CLAIM_SIZE=$(kubectl get pvc -n "$MINIO_NAMESPACE" minio-pv-claim --no-headers -o custom-columns=:.spec.resources.requests.storage)
```

Comment:
we have a pattern of using -o jsonpath or --template. can you use that instead?

Reply:
fixed.